rfcs/rdm-0064-custom-fields.md

```diff
@@ -0,0 +1,440 @@
+
+### Use Cases
+
+**As an instance administrator...**
+
+- I want to add a custom fields to the record's data model, and/or the community's data model.
+- I want users to be able to input values for those custom fields via the upload and community creation forms.
+- I want to add custom fields for internal use only (i.e. not visible to the users, nor available in the forms).
+- I want to be able to hide some fields in some communities. While the fields might still be present in the data model, they won't be shown/available for input. Alternatively, I want to be able to configure custom fields per community.
+
+**The following use cases apply to all the above points**, and focus on the how fields are defined and used.
+
+- I want to be able to define the type: primitive (string, number, date), complex (list, dictionary), and vocabularies (both generic and specific).
+- I want to be able to define required fields.
+
+**When displaying the custom fields in the forms...**
+
+- I want to define groups for my new fields.
+- I want to define in which order they will display inside those groups.
+- I want to add fields to already existing groups (e.g. Basic information).
+- I want to support translations (i18n) of the fields' label and description.
+
+**When displaying the custom fields of records...**
+
+- I want to define where and how the field will be shown in the landing/community page.
+- I want to define how the values will be serialized into different formats (e.g. Dublin Core, Darwin Core, DataCite).
+- I want to be able to filter my search results based on my new field.
+- I want to support translations (i18n) for the values of the custom fields.
+
+### Fist iteration limitations
+
+The first iteration of custom fields will be available in InvenioRDM v10. It targets the implementation of two vertical workflows, allowing to add free text custom fields (e.g. keywords, descriptions) and also based on generic vocabularies (e.g. defining a new "experiment" vocabulary). This will enable the implementation of the mechanisms to support custom fields, and allow developers to extend support for new types of fields (e.g. dates).
+
+Therefore, it **comes with some limitation**:
+
+- Only two field types are supported: Text and generic vocabularies. For example, resource types, or new ones of your creation that follow the `VocabularySchema`. This means, adhering to the following structure:
+
+```yaml
+id: ...
+icon: ...
+props: ...
+description: ...
+tags: ...
+title: ...
+```
+
+- Fields are configured globally for all records and for all communities. Note that they do not mix, those appearing in records will not appear in communities. However, all communities will have the same fields.
+- You can hide (not display) fields in the UI. However, they will be available in the REST API. Moreover, hiding them will hide them globally (i.e. for all records or communities).
```

Comment:
Can you specify if this is both for the upload/settings forms and for the details page of record/community, or only for one of them?

Reply:
Nice catch, they will be displayed in the generic view if given via REST API. It's kind of a "hiding by obscurity".